Buildings/Controls/OBC/CDL/Logical/Sources/TimeTable.mo

```diff
@@ -0,0 +1,235 @@
+  Continuous.Abs abs1[nout]
+    "Absolute value"
+    annotation (Placement(transformation(extent={{-60,-100},{-40,-80}})));
+  Continuous.GreaterThreshold greThr[nout](
+    final t=fill(Constants.small, nout))
+    "Value comparisson"
+    annotation (Placement(transformation(extent={{-20,-100},{0,-80}})));
+  Not not1
+    "Logical not"
+    annotation (Placement(transformation(extent={{60,-100},{80,-80}})));
+
+initial equation
+  t0=time;
+
+equation
+  assert(n > 0, "No table values defined.");
+  assert(extrapolation <> CDL.Types.Extrapolation.LastTwoPoints, "Unsuitable extrapolation setting.");
+
+  connect(tab.y, realToInteger.u)
+    annotation (Line(points={{-79,0},{-62,0}}, color={0,0,127}));
+  connect(realToInteger.y, intEquFal.u2) annotation (Line(points={{-38,0},{-30,0},
+          {-30,30},{-50,30},{-50,72},{-42,72}}, color={255,127,0}));
+  connect(realToInteger.y, intEquTru.u2) annotation (Line(points={{-38,0},{-30,0},
+          {-30,30},{-50,30},{-50,42},{-42,42}}, color={255,127,0}));
+  connect(intFal.y, intEquFal.u1)
+    annotation (Line(points={{-58,80},{-42,80}}, color={255,127,0}));
+  connect(intTru.y, intEquTru.u1)
+    annotation (Line(points={{-58,50},{-42,50}}, color={255,127,0}));
+  connect(intEquFal.y, or2.u1) annotation (Line(points={{-18,80},{0,80},{0,70},{
+          18,70}}, color={255,0,255}));
+  connect(intEquTru.y, or2.u2) annotation (Line(points={{-18,50},{0,50},{0,62},{
+          18,62}}, color={255,0,255}));
+  connect(or2.y, mulAnd.u)
+    annotation (Line(points={{42,70},{48,70}}, color={255,0,255}));
+  connect(intEquTru.y, y) annotation (Line(points={{-18,50},{0,50},{0,0},{140,0}},
+        color={255,0,255}));
+  connect(realToInteger.y, intToRea.u) annotation (Line(points={{-38,0},{-30,0},
+          {-30,-20},{-22,-20}}, color={255,127,0}));
+  connect(intToRea.y, add2.u1) annotation (Line(points={{2,-20},{10,-20},{10,-44},
+          {18,-44}}, color={0,0,127}));
+  connect(tab.y, add2.u2) annotation (Line(points={{-79,0},{-70,0},{-70,-56},{18,
+          -56}}, color={0,0,127}));
+  connect(add2.y, abs1.u) annotation (Line(points={{42,-50},{50,-50},{50,-70},{-70,
+          -70},{-70,-90},{-62,-90}}, color={0,0,127}));
+  connect(abs1.y, greThr.u)
+    annotation (Line(points={{-38,-90},{-22,-90}}, color={0,0,127}));
+  connect(greThr.y, mulOr1.u)
+    annotation (Line(points={{2,-90},{18,-90},{18,-90}}, color={255,0,255}));
+  connect(mulOr1.y, not1.u)
+    annotation (Line(points={{42,-90},{58,-90}}, color={255,0,255}));
+  connect(not1.y, and2.u2) annotation (Line(points={{82,-90},{90,-90},{90,-20},{
+          50,-20},{50,22},{58,22}}, color={255,0,255}));
+  connect(mulAnd.y, and2.u1) annotation (Line(points={{72,70},{80,70},{80,50},{50,
+          50},{50,30},{58,30}}, color={255,0,255}));
+  connect(and2.y, assMes1.u) annotation (Line(points={{82,30},{90,30},{90,70},{98,
+          70}}, color={255,0,255}));
+  annotation (
```

Comment:
That looks very complicated. Why not just use the output of the table with a `Less` block to convert to a boolean?

Reply:
Would that be `Less` than 0.5 that you are suggesting? How does one make sure that the user has not entered values below zero, very large values, values between 0 and 1?